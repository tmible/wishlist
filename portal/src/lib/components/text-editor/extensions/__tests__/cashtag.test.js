import { describe, expect, it, vi } from 'vitest';
import Cashtag from '../cashtag.js';

describe('text editor / extensions / cashtag', () => {
  it('should be defined', () => {
    expect(Cashtag).toMatchSnapshot();
  });

  it('should parse HTML', () => {
    expect(Cashtag.config.parseHTML()).toMatchSnapshot();
  });

  it('should render HTML', () => {
    expect(Cashtag.config.renderHTML()).toMatchSnapshot();
  });

  describe('add commands', () => {
    it('should add set command', () => {
      const setMark = vi.fn();
      Cashtag.config.addCommands().setCashtag()({ commands: { setMark } });
      expect(setMark).toHaveBeenCalledWith(Cashtag.name);
    });

    it('should add toggle command', () => {
      const toggleMark = vi.fn();
      Cashtag.config.addCommands().toggleCashtag()({ commands: { toggleMark } });
      expect(toggleMark).toHaveBeenCalledWith(Cashtag.name);
    });

    it('should add unset command', () => {
      const unsetMark = vi.fn();
      Cashtag.config.addCommands().unsetCashtag()({ commands: { unsetMark } });
      expect(unsetMark).toHaveBeenCalledWith(Cashtag.name);
    });
  });

  it('should add input rules', () => {
    expect(Cashtag.config.addInputRules()).toMatchSnapshot();
  });

  it('should add paste rules', () => {
    expect(Cashtag.config.addPasteRules()).toMatchSnapshot();
  });
});
