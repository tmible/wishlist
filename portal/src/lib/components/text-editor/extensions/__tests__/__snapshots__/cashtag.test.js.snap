// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text editor / extensions / cashtag > should add input rules 1`] = `
[
  InputRule {
    "find": /\\(\\?:\\^\\|\\\\s\\)\\(\\(\\\\\\$\\[A-Z\\]\\{3,8\\}\\)\\\\s\\)\\$/,
    "handler": [Function],
  },
]
`;

exports[`text editor / extensions / cashtag > should add paste rules 1`] = `
[
  PasteRule {
    "find": /\\(\\?:\\^\\|\\\\s\\)\\(\\(\\\\\\$\\[A-Z\\]\\{3,8\\}\\)\\\\s\\)/g,
    "handler": [Function],
  },
]
`;

exports[`text editor / extensions / cashtag > should be defined 1`] = `
Mark {
  "child": null,
  "config": {
    "addCommands": [Function],
    "addInputRules": [Function],
    "addPasteRules": [Function],
    "defaultOptions": {},
    "name": "cashtag",
    "parseHTML": [Function],
    "renderHTML": [Function],
  },
  "name": "cashtag",
  "options": {},
  "parent": null,
  "storage": {},
  "type": "mark",
}
`;

exports[`text editor / extensions / cashtag > should parse HTML 1`] = `
[
  {
    "tag": "span[data-entity="cashtag"]",
  },
]
`;

exports[`text editor / extensions / cashtag > should render HTML 1`] = `
[
  "span",
  {
    "class": "link cursor-text",
  },
  0,
]
`;
